src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TseClaimantRepReplyService.java

```diff
@@ -0,0 +1,366 @@
+            return uk.gov.hmcts.ecm.common.helpers.DocumentHelper.claimantApplicationTypeToDocType(
+                    claimantApplicationType);
+        }
+    }
+
+    /**
+     * Reply to a TSE application as a respondent, including updating app status, saving the reply and sending emails.
+     *
+     * @param userToken authorisation token to get claimant's email address
+     * @param caseDetails case details
+     * @param caseData case data
+     */
+    public void claimantReplyToTse(String userToken, CaseDetails caseDetails, CaseData caseData) {
+        updateApplicationState(caseData);
+
+        boolean isRespondingToTribunal = isRespondingToTribunal(caseData);
+        saveReplyToApplication(caseData, isRespondingToTribunal);
+
+        if (isRespondingToTribunal) {
+            sendRespondingToTribunalEmails(caseDetails, userToken);
+        } else {
+            sendRespondingToApplicationEmails(caseDetails, userToken);
+        }
+
+        resetReplyToApplicationPage(caseData);
+    }
+
+    /**
+     * Change status of application to updated if there was an unanswered request for information from the admin.
+     *
+     * @param caseData in which the case details are extracted from
+     */
+    void updateApplicationState(CaseData caseData) {
+        GenericTseApplicationType selectedApplicationType = getClaimantRepSelectedApplicationType(caseData);
+        if (selectedApplicationType.getApplicant().equals(RESPONDENT_TITLE)) {
+            if (isRespondingToTribunal(caseData)) {
+                selectedApplicationType.setApplicationState(WAITING_FOR_THE_TRIBUNAL);
+            } else {
+                selectedApplicationType.setApplicationState(UPDATED);
+            }
+        } else if (isRespondingToTribunal(caseData)) {
+            selectedApplicationType.setApplicationState(UPDATED);
+        }
+    }
+
+    /**
+     * Saves the data on the reply page onto the application object.
+     *
+     * @param caseData contains all the case data
+     * @param isRespondingToTribunal determines if responding to the Tribunal's request/order
+     */
+    void saveReplyToApplication(CaseData caseData, boolean isRespondingToTribunal) {
+        GenericTseApplicationType genericTseApplicationType = getRespondentSelectedApplicationType(caseData);
+
+        if (isEmpty(genericTseApplicationType.getRespondCollection())) {
+            genericTseApplicationType.setRespondCollection(new ArrayList<>());
+        }
+        List<TseRespondTypeItem> respondCollection = genericTseApplicationType.getRespondCollection();
+
+        TseRespondType response = TseRespondType.builder()
+                .response(caseData.getTseResponseText())
+                .supportingMaterial(caseData.getTseResponseSupportingMaterial())
+                .hasSupportingMaterial(caseData.getTseResponseHasSupportingMaterial())
+                .from(RESPONDENT_TITLE)
```

Comment:
Should this be Claimant or Claimant Rep?

Reply:
this is the response from respondent to claimant application. so it is RESPONDENT_TITLE

Reply:
This line from the TseClaimantRepReplyController seems to suggest that this is the claimant reply replying?

https://github.com/hmcts/et-ccd-callbacks/pull/2259/files#diff-1a46e9643d91bee8708c4fe3c19f0450832243a48df6e297359f1bc45a161da7R198